src/radical/pilot/umgr/staging_input/default.py

```diff
@@ -338,12 +340,23 @@ def _handle_unit(self, unit, actionables):
             src = complete_url(src, src_context, self._log)
             tgt = complete_url(tgt, tgt_context, self._log)
 
-            if rpc.CREATE_PARENTS in flags:
-                copy_flags = rs.filesystem.CREATE_PARENTS
-            else:
-                copy_flags = 0
+            # Check if the src is a folder, if true
+            # add recursive flag if not already specified
+            if os.path.isdir(src.path):
+                if rs.filesystem.RECURSIVE not in flags:
+                    flags += [rs.filesystem.RECURSIVE]
 
-            saga_dir.copy(src, tgt, flags=copy_flags)
+            # Check if create parents flag is set,
+            # if not set it.
+            if rs.filesystem.CREATE_PARENTS not in flags:
+                flags += [rs.filesystem.CREATE_PARENTS]
+
+            if flags:
```

Comment:
You just made sure that flags is never empty.  You can also remove the special case of an empty list with:
```
>>> flags = list()
>>> reduce(lambda x, y: x | y, flags, 0)
0
```

Reply:
The above was to enforce that CREATE_PARENTS is always there by default(and cannot be overwritten by the user). That means the list will never be empty. If you think that is unnecessary, let me know.